### ISSUE TITLE
get_clock_time() method 

### ISSUE THREAD

**Issue by VolkrB** [sandialabs/pecos#30]
Hey! 
I've been setting my time filter without using this method, because it tends to be pretty slow (especially for large data). For the simple example timefilter the timeit on my system is:
%timeit  298ms per loop

My work around is to set my time filter using the datetime library and df.index.time. I could do a pull-request illustrating this in an example, if you are interested. My timeit for the simple example timefilter is:
%timeit 6.55ms per loop

Volker

**Comment by kaklise** [comment]
That sounds great!  I look forward to seeing your code.

**Comment by adriesse** [comment]
I looked into the code out of curiosity.  If you're going to make changes to this function, I suggest making the description a bit more specific also, for example:

"Determine the time of day in seconds past midnight for each Timestamp in the DataFrame's DatetimeIndex."

**Comment by wholmgren** [comment]
I also looked out of curiosity. Can you can replace the slow loop with array operations like this?

```python
In [10]: index = pd.DatetimeIndex(start='20160926', end='20160928', freq='1h')

In [11]: secofday = index.hour*3600 + index.minute*60 + index.second + index.microsecond/1000000

In [12]: secofday = pd.Series(secofday, index=index)
```

This code works, but maybe I'm missing something.

**Comment by kaklise** [comment]
Both options make the code much faster.  I tested this out on large synthetic data, and the method from @VolkrB is at least 3 orders of magnitude faster than the previous method.  When I removed the loop, getting clock time and creating a time filter was even faster.  Thanks for catching this bottle neck in the code.  I'll check in changes that remove the loop from 'get_clock_time' and update the documentation.

**Comment by kaklise** [comment]
Closed with pull request #32

**Comment by adriesse** [comment]
Just for fun, here is another way that seems to work:
```
In [3]: index = pd.DatetimeIndex(start='20160926', end='20160928', freq='1h')

In [4]: dates = pd.DatetimeIndex(index.date)

In [5]: times = pd.TimedeltaIndex(index.values-dates.values)

In [6]: secofday = pd.Series(times.total_seconds(), index=index)
```
